sway/commands/move.c

```diff
@@ -271,12 +271,11 @@ static void workspace_rejigger(struct sway_workspace *ws,
 		return;
 	}
 	container_detach(child);
-	struct sway_container *new_parent = workspace_wrap_children(ws);
+	workspace_wrap_children(ws);
 
 	int index =
 		move_dir == WLR_DIRECTION_LEFT || move_dir == WLR_DIRECTION_UP ? 0 : 1;
 	workspace_insert_tiling(ws, child, index);
-	container_flatten(new_parent);
 	ws->layout =
```

Comment:
We shouldn't be doing this flattening: if we had `T[app app]` to begin with, by now we have `H[T[app] T[app]]`. If we flatten the old workspace, we end up with `H[app T[app]]`, which is incorrect.

Reply:
`git blame` shows that this flattening was added in https://github.com/swaywm/sway/pull/2695. However, the two test cases in that PR continue to work now even without this flattening.